codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -218,13 +218,8 @@ private void generateOperationSerializerMiddleware(GenerationContext context, Op
 
             writer.write("");
             writer.write("opPath, opQuery := httpbinding.SplitURI($S)", httpTrait.getUri());
-            writer.write("request.URL.Path = opPath");
-            writer.openBlock("if len(request.URL.RawQuery) > 0 {", "", () -> {
-                writer.write("request.URL.RawQuery = \"&\" + opQuery");
-                writer.openBlock("} else {", "}", () -> {
-                    writer.write("request.URL.RawQuery = opQuery");
-                });
-            });
+            writer.write("request.URL.Path = smithyhttp.JoinPath(request.URL.Path, opPath)");
+            writer.write("request.URL.RawQuery = smithyhttp.JoinRawQuery(request.URL.RawQuery, opQuery)");
```

Comment:
Nit: Should import `SmithyGoDependency.SMITHY_HTTP_TRANSPORT`

Reply:
This is done above this section, on line 200.

Reply:
No - line 200, has `SMITHY_HTTP_BINDING`, which imports `httpbinding`.  We should be importing `SmithyGoDependency.SMITHY_HTTP_TRANSPORT`